src/refdb_fs.c

```diff
@@ -63,6 +66,18 @@ typedef struct refdb_fs_backend {
 	uint32_t direach_flags;
 } refdb_fs_backend;
 
+/*
+ * List of symbolic reference names that are always stored in the
+ * gitdir. This is required such that we are able to correctly
+ * resolve these references via the gitdir instead of the
+ * commondir when working trees are used.
+ */
+static const char *symrefs[] = {
```

Comment:
Maybe something like `uncommon_refs`, as those would not be stored in the common directory. Not exactly happy with this name, though, but I'd prefer either this or `workdir_refs`.